### DISCUSSION TITLE
Supporting inverted-y tile coords in CRS.Simple (or similar)

### DISCUSSION THREAD

**Discussion by BrianOfTsushima** [Leaflet/Leaflet#8393] [General]
We use leaflet to render maps from video games.  In our case (seems typical for many games) we use a conventional "Z-Up" world coordinate system for games with the origin at the center of the map, and +x is right/east and +y is up/north.

For a long while we've used CRS.Simple successfully and used a huge imageOverlay bitmap for our map.

Recently, we decided to add detail to our map, and in doing so converted to a proper tiling map.  However, the coordinate system for tile names  -- even when using CRS.Simple, has +y going down/south.  The {-y} implementation requires bounds *and* implicitly presumes the map resides in the +x / +y quadrant.  In our case, with our map centered at the origin, this is untrue.

Happily, with very modest changes, it's straightforward to offer an alternate approach -- in cases where the coordinate system is infinite, it seems correct (at least to me) to offer {-y} as simply the inverted coord.y minus 1.  (minus 1 moves origin to bottom left of tile 0,0, which is what I believe most/all will expect).   It's a modest addition to TileLayer::getTileUrl, which I'd be happy to submit, if this seems sane?

The bonus of this approach (IMO!) is it will only enable additional cases where {-y} is provided --meaning the impact to existing users should be zero.

**Comment by IvanSanchez** [comment]
You're right - using `{-y}` in tiles needs the CRS to be finite - I remember from https://stackoverflow.com/questions/62622737 .

This sounds sane, but it's better suited as a feature request under "issues" rather than a "discussion". A change in the `GridLayer` logic to support `{-y}` in infinite CRSs (without breaking anything!) shouldn't be controversial.

**Comment by BrianOfTsushima** [comment]
Ok, I'll make it there, feel free to close this

**Reply by IvanSanchez** [reply]
#8395, I see.
